src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -211,6 +211,8 @@ internal class SqlServerFhirDataStore : IFhirDataStore, IProvideCapability
 
                     stream.Seek(0, 0);
 
+                    _logger.LogInformation($"Upserting {resource.ResourceTypeName} with a stream length of {stream.Length}");
```

Comment:
We shouldn't use string interpolation with structured logging:
e.g.
```
_logger.LogInformation("Upserting {ResourceTypeName} with a stream length of {StreamLength}", resource.ResourceTypeName, stream.Length);
```

Reply:
Based on what we know now, this won't log the cases where rawResources are saved with 0 length. The length is ok,  its that the stream has already been read (so its at the end) on the subsequent retries.